src/itehandler/IteToSwitch.cc

```diff
@@ -0,0 +1,283 @@
+//
+// Created by prova on 04.08.20.
+//
+
+#include "IteToSwitch.h"
+#include "Logic.h"
+
+ite::Node::Node(PTRef term, PTRef cond, NodeRef true_node, NodeRef false_node, uint32_t id)
+    : id(id)
+    , type(cond != PTRef_Undef ? static_cast<bool>(NodeType::inner) : static_cast<bool>(NodeType::leaf))
+    , term(term)
+{
+    if (cond != PTRef_Undef) {
+        // a leaf node.
+        setCond(cond);
+        setTrueChild(true_node);
+        setFalseChild(false_node);
+    }
+}
+
+ite::NodeRef ite::Dag::newNode(PTRef val) {
+    assert(not nodes.has(val));
+    nodeRefs.push(na.alloc(val, PTRef_Undef, NodeRef_Undef, NodeRef_Undef));
+    nodes.insert(val, nodeRefs.last());
+    return nodeRefs.last();
+}
+
+ite::NodeRef ite::Dag::newNode(PTRef tr, PTRef cond, ite::NodeRef true_node, ite::NodeRef false_node)
+{
+    assert(not nodes.has(tr));
+    nodeRefs.push(na.alloc(tr, cond, true_node, false_node));
+    nodes.insert(tr, nodeRefs.last());
+    return nodeRefs.last();
+}
+
+void ite::Dag::annotateWithParentInfo(PTRef root_tr) {
+    NodeRef root = getNode(root_tr);
+    vec<NodeRef> queue;
+    vec<type> flag;
+    flag.growTo(na.getNumNodes());
+    parents[root] = ParentSet();
+    queue.push(root);
+    while (queue.size() != 0) {
+        NodeRef el_r = queue.last();
+        const Node &el = na[el_r];
+        if (flag[el.getId()] == type::black) {
+            queue.pop();
+            continue;
+        }
+        flag[el.getId()] = type::gray;
+
+        bool unprocessed_children = false;
+
+        for (lbool childPolarity : { l_False, l_True }) {
+            NodeRef child = childPolarity == l_False ? el.getFalseChild() : el.getTrueChild();
+            if (child != NodeRef_Undef) {
+                if (parents.find(child) == parents.end()) {
+                    parents[child] = ParentSet();
+                }
+                parents[child].insert({el_r, childPolarity});
```

Comment:
The check 
```
if (parents.find(child) == parents.end()) {
     parents[child] = ParentSet();
}
```
is not necessary. The operator[] on std::map takes care of that, since it always try to insert the key with default-constructed value.